### PR TITLE
Async loading

### PR DESCRIPTION
**Pull request by glebm**
Does the loading on a separate thread, so we're now loading while fading in and while updating the progress bar (resulting in much faster loads; fade outs still happen only after loading).

I had to separate the virtual gamepad surface and texture (de)initialization code because it was logging error messages, and upon further investigation I learned that per https://wiki.libsdl.org/SDL3/CategoryRender:
> These functions must be called from the main thread. See this bug for details: https://github.com/libsdl-org/SDL/issues/986

I also had to adjust fatal error reporting: `app_fatal` can only be called from the main thread, so during loading we instead propagate a `tl::expected` and send a custom message to the main thread if there is an error. Introduces `RETURN_IF_ERROR` and `ASSIGN_OR_RETURN` macros for more ergonomic status propagation.

I think I found most of the places where we currently call `app_fatal` during load and replaced them with `tl::expected` propagation. If I missed any, we won't see an error dialog for these errors and the loading process will appear to hang with lots of console spam instead.

Also adds an advanced (ini-only) setting to skip showing loading screen on fast loads. This is what it looks like on my machine with the threshold set to 500ms.

https://github.com/user-attachments/assets/37ff60ca-a247-4e06-b7e3-f5b6b8ea3692

### CODE REVIEW

File: Source/utils/ini.cpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ tl::expected<Ini, std::string> Ini::parse(std::string_view buffer)
 			nextLineBegin = eof;
 		} else {
 			nextLineBegin = lineEnd + 1;
-			if (lineBegin + 1 < lineEnd && *(lineEnd - 1) == '\r') --lineEnd;
+			if (lineBegin + 1 <= lineEnd && *(lineEnd - 1) == '\r') --lineEnd;
```
Comment (qndel) [inline Source/utils/ini.cpp:120]: doesn't seem related :)

Comment (glebm) [reply Source/utils/ini.cpp:120]: It's in a separate commit that I also sent as another PR

File: Source/controls/touch/renderers.cpp [hunk 2]
```diff
@@ -516,30 +491,100 @@ void VirtualGamepadRenderer::UnloadArt()
...
+	Renderer.LoadArt();
 }
 
 void FreeVirtualGamepadGFX()
 {
 	Renderer.UnloadArt();
 }
 
+#ifndef USE_SDL1
+void VirtualGamepadRenderer::createTextures(SDL_Renderer &renderer)
```
Comment (StephenCWills) [inline Source/controls/touch/renderers.cpp:521]: These files don't get compiled in with SDL1.
https://github.com/diasurgical/devilutionX/blob/d8dde8d2b119dfbc7973384aa598bd9e9de2c864/Source/CMakeLists.txt#L254-L260

The only reason there's a distinction between texture rendering vs surface rendering is because of the `Upscale` option.

Comment (glebm) [reply Source/controls/touch/renderers.cpp]: Ah, thanks! Done

**Comment by AJenbo** [comment]
Hail Marry!

(don't have time to fully review, but it's a good time for me to test it with others)
